---
int32 jnt_num